### PR TITLE
perf: Add criterion benchmark for aggregate expressions

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

Add Criterion benchmark to compare DataFusion and Comet aggregate expressions.

```
aggregate/avg_decimal_datafusion
                        time:   [653.56 µs 657.57 µs 662.06 µs]
aggregate/avg_decimal_comet
                        time:   [1.0581 ms 1.0592 ms 1.0604 ms]
aggregate/sum_decimal_datafusion
                        time:   [695.51 µs 696.48 µs 697.60 µs]
aggregate/sum_decimal_comet
                        time:   [1.0218 ms 1.0230 ms 1.0242 ms]
```

## What changes are included in this PR?

New benchmark

## How are these changes tested?

Manually

### CODE REVIEW

File: native/core/Cargo.toml [hunk 1]
```diff
@@ -122,3 +122,7 @@ harness = false
 [[bench]]
 name = "filter"
 harness = false
+
+[[bench]]
+name = "aggregate"
+harness = false
```
Comment (huaxingao) [inline native/core/Cargo.toml:128]: nit: shall we add an empty line to the end of the file?

**Review by huaxingao — APPROVED** [review]
LGTM

File: native/core/benches/aggregate.rs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                c0.clone(),
+                c1.clone(),
+                datafusion_sum_decimal.clone(),
+                "sum",
+            )
+        })
+    });
+
+    group.bench_function("sum_decimal_comet", |b| {
+        let comet_sum_decimal = Arc::new(AggregateUDF::new_from_impl(SumDecimal::new(
```
Comment (kazuyukitanimura) [inline native/core/benches/aggregate.rs:99]: do we need to use blackbox to avoid the optimization?

**Review by kazuyukitanimura — APPROVED** [review]

**Review by viirya — APPROVED** [review]

File: native/core/benches/aggregate.rs (hunk 2 above)
Comment (andygrove) [reply native/core/benches/aggregate.rs:99]: added, thanks

**Pull request merged**